tools/mock-service-host/src/mid/oav/swaggerMocker.ts

```diff
@@ -125,11 +130,16 @@ export default class SwaggerMocker {
             const ret: Record<string, any> = {}
             // eslint-disable-next-line prefer-const
             for (let [key, item] of Object.entries(obj)) {
-                if (inUserAssignedIdentities) {
+                if (
+                    inUserAssignedIdentities &&
+                    !key.match(
+                        /\/subscriptions\/.*\/providers\/Microsoft.ManagedIdentity\/userAssignedIdentities\/.*/i
+                    )
+                ) {
                     const subscription =
                         pathElements.subscriptions || '00000000-0000-0000-0000-000000000000'
                     const resourceGroup = pathElements.subscriptions || 'mockGroup'
-                    key = `/subscriptions/${subscription}/resourceGroups/${resourceGroup}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/${key}`
+                    key = `/subscriptions/${subscription}/resourceGroups/${resourceGroup}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/aa`
```

Comment:
why do we need 'aa' for the identity value?

Reply:
I think here we just want to mock a valid userAssignedIdentity, I changed "aa" to "mocked" to make it more readable. 
The original behavior trapped if there is special characters in `${key}`